File: src-old/test/java/io/tripled/social/client/presentation/controller/PostControllerTest.java
package io.tripled.social.client.presentation.controller;

import io.tripled.social.client.application.SpiedPostMessageUseCase;
import io.tripled.social.client.presentation.TestOutput;
import io.tripled.social.client.presentation.request.PostCliRequest;
import org.junit.Before;
import org.junit.Test;

import static org.hamcrest.Matchers.equalTo;
import static org.junit.Assert.assertThat;

public class PostControllerTest {

  private PostController postController;
  private SpiedPostMessageUseCase service;

  @Before
  public void setUp() {
    service = new SpiedPostMessageUseCase();
    postController = new PostController(service);
  }

  @Test
  public void executesPostCommand() {
    PostCliRequest request = new PostCliRequest("Alice", "Hi There!");


    postController.execute(request, new TestOutput());

    assertThat(service.userName, equalTo("Alice"));
    assertThat(service.message, equalTo("Hi There!"));
  }

}

File: src/test/kotlin/io/tripled/social/client/infrastructure/cli/controller/PostControllerTest.kt
package io.tripled.social.client.infrastructure.cli.controller

import io.tripled.social.client.application.usecase.spies.PostMessageSpy
import io.tripled.social.client.infrastructure.cli.TestOutput
import io.tripled.social.client.infrastructure.cli.request.PostCliRequest
import org.hamcrest.MatcherAssert
import org.hamcrest.Matchers
import org.junit.jupiter.api.BeforeEach
import org.junit.jupiter.api.Test

internal class PostControllerTest {
    private var postController: PostController? = null
    private var service: PostMessageSpy? = null

    @BeforeEach
    fun setUp() {
        service = PostMessageSpy()
        postController = PostController(service!!)
    }

    @Test
    fun executesPostCommand() {
        val request = PostCliRequest("Alice", "Hi There!")


        postController!!.execute(request, TestOutput())

        MatcherAssert.assertThat<String?>(service!!.userName, Matchers.equalTo<String?>("Alice"))
        MatcherAssert.assertThat<String?>(service!!.message, Matchers.equalTo<String?>("Hi There!"))
    }
}